scripts/cmake/xtensa-toolchain.cmake

```diff
@@ -50,8 +53,10 @@ set(CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY)
 if(TOOLCHAIN STREQUAL "xt")
 	set(XCC 1)
 	set(CMAKE_C_COMPILER ${CROSS_COMPILE}xcc)
+	set(CMAKE_CXX_COMPILER ${CROSS_COMPILE}xc++)
 else()
 	set(CMAKE_C_COMPILER ${CROSS_COMPILE}gcc)
+	set(CMAKE_CXX_COMPILER ${CROSS_COMPILE}gcc)
```

Comment:
IIUC GCC will build in C++ mode if invoked as ```g++``` otherwise it could build headers as C